frontend/common/AnsibleLogin.tsx

```diff
@@ -0,0 +1,178 @@
+import {
+  BackgroundImage,
+  Brand,
+  Login,
+  LoginForm,
+  LoginHeader,
+  LoginMainBody,
+  LoginMainFooter,
+  LoginMainHeader,
+} from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import { ReactNode, useCallback, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ErrorBoundary } from '../../framework/components/ErrorBoundary';
+import { useFrameworkTranslations } from '../../framework/useFrameworkTranslations';
+import { AuthOption, SocialAuthLogin } from './SocialAuthLogin';
+import { RequestError, createRequestError } from './crud/RequestError';
+import { getCookie } from './crud/cookie';
+
+export function AnsibleLogin(props: {
+  /** Title for the login main body header of the login page */
+  loginTitle?: string;
+
+  /** Subtitle for the login main body header of the login page */
+  loginSubtitle?: string;
+
+  /** The brand image for the login page */
+  brandImg?: ReactNode;
+
+  /** Attribute that specifies the alt text of the brand image for the login page */
+  brandImgAlt: string;
+
+  /** Content rendered inside of the text component of the login page */
+  textContent?: string;
+
+  /** Content rendered inside of social media login footer section */
+  authOptions?: AuthOption[];
+
+  /** Attribute that specifies the URL of the background image for the login page */
+  backgroundImgSrc?: string;
+
+  /** The url to the API endpoint for logging in */
+  loginApiUrl: string;
+
+  /** Callback function that is called when the user successfully logs in */
+  onSuccess: () => void;
+}) {
+  const { t } = useTranslation();
+  const [translations] = useFrameworkTranslations();
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [helperText, setHelperText] = useState('');
+
+  const { loginApiUrl } = props;
+  const onSubmit = useCallback(async () => {
+    try {
+      const loginPageResponse = await fetch(loginApiUrl, {
+        credentials: 'include',
+        headers: { Accept: 'application/json,text/*' },
+      });
+      if (!loginPageResponse.ok) {
+        throw await createRequestError(loginPageResponse);
```

Comment:
Previously, we were using `LoginForm` and `GenericForm`, and any errors raised would bubble up to these components. `GenericForm` would then parse the raised errors and display them to the user. Now, with the use of PatternFly (PF) native components, I'm not sure if raising this error here will still result in it being displayed to the user. I believe we are relying on the ErrorBoundary now.
 @jamestalton, what do you think?

Reply:
The PF component has the ability to show the error text is in a standard way across products.
We wrap the submit in a try/catch and then set the error test with setHelperText(err.message).
The error boundary should not be catching/handling any errors.